MapboxCoreNavigation/RouteController.swift

```diff
@@ -719,23 +721,31 @@ extension RouteController: CLLocationManagerDelegate {
                 
                 // Look at the following step to determine what the new alert level should be
                 if let upComingStep = routeProgress.currentLegProgress.upComingStep {
-                    alertLevel = upComingStep.expectedTravelTime <= RouteControllerMediumAlertInterval ? .medium : .low
+                    alertLevel = calculateNewAlert(from: upComingStep)
                 } else {
                     assert(false, "In this case, there should always be an upcoming step")
                 }
             }
-        } else if secondsToEndOfStep <= RouteControllerHighAlertInterval && routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForHighAlert {
```

Comment:
As far as I can tell, the plans are to have the Directions API provide a distance ahead of the maneuver, not a duration. It seems like deemphasizing the time-based heuristics in favor of the distance thresholds would be more future-proof in that case.